src/dataqueue/queue.cc

```diff
@@ -784,21 +792,194 @@ class DataQueueEntry : public EntryBase {
 
 // ============================================================================
 
-// TODO(@flakey5): Implement StreamEntry, StreamBaseEntry, and FdEntry
-// These will all extend from DataQueueEntry in various ways. The StreamEntry
-// and StreamBaseEntry will each always be non-idempotent. Reads will always
-// be destructive of the state. FdEntry, although it is streaming, can be
-// idempotent or non-idempotent, depending on the file descriptor type.
+// Essentially an entry that exists to give the Javascript side
+//  control of what happens when data is read. Always non-idempotent.
+class StreamEntry : public BaseObject, public EntryBase {
+ public:
+  static void New(const FunctionCallbackInfo<Value>& args) {
+    CHECK(args.IsConstructCall());
+    Environment* env = Environment::GetCurrent(args);
+
+    CHECK(args.Length() == 1);
+
+    const Local<Value>& pullCallback = args[0];
+    CHECK(pullCallback->IsFunction());
+
+    new StreamEntry(env, args.This(), pullCallback);
```

Comment:
This can simply be:

```suggestion
    CHECK(args[0]->IsFunction());

    new StreamEntry(env, args.This(), args[0]);
```

No reason to define a separate const ref.